Assets/Scripts/KatamariMovement.cs

```diff
@@ -50,24 +64,109 @@ void Update()
         verticalInput = Input.GetAxis("Vertical");
         hitInput = Input.GetAxis("Jump");
         stopInput = Input.GetAxis("Stop");
-        isGolfHitMode = Input.GetAxis("Mode") > 0.5;
 
-        hitXAngle += hitXAngleSpeed * verticalInput * Time.deltaTime;
+
+
+
+        // turn the power bar on and off depending if we can hit hit the ball
+        if (power_bar_active)
+        {
+            power_bar.SetActive(true);
+            isGolfHitMode = true;
+            this.GetComponent<LineRenderer>().enabled = true;
+
+
+        }
+        else
+        {
+            power_bar.SetActive(false);
+            this.GetComponent<LineRenderer>().enabled = false;
+        }
+
+        if (resourceManager.can_hit())
+        {
+            power_bar_active = true;
+        }
+
+
+        // SLOW MOTION
+
+        if (power_bar_active == false && hitInput > 0.5 && resourceManager.GetComponent<ResourceManager>().GetTimeSinceLastHit() >= 0.2f)
```

Comment:
There's some duplication of checking the time since last hit that would be good to clean up. Also, we shouldn't need the `GetComponent` since we are already referencing the `ResourceManager`:
```suggestion
        float flyMovementDeadTime = 0.2f;
        bool isFlyMovementDeadTime = resourceManager.GetTimeSinceLastHit() >= flyMovementDeadTime;
        if (power_bar_active == false && hitInput > 0.5 && isFlyMovementDeadTime)
```